src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/prisonapi/PrisonerApiService.kt

```diff
@@ -76,18 +77,43 @@ internal class PrisonerApiService(
         }
 
         val offender = prisonApiClient.retrieveOffender(nomsId).block()
-        sentencesAndOffences.copy(
+        sentenceAndOffences.copy(
           releaseDate = movement?.dateOutOfPrison,
           releasingPrison = prisonDescription,
           licenceExpiryDate = offender?.sentenceDetail?.licenceExpiryDate,
-          offences = sentencesAndOffences.offences.sortedBy { it.offenceDescription },
+          offences = sentenceAndOffences.offences.sortedBy { it.offenceDescription },
         )
       }
+        .filter { it.sentenceEndDate == null || !it.sentenceEndDate.isBefore(LocalDate.now()) }
+        .sortedBy { it.sentenceSequence }
+
+      // Sequencing happens here
+      val sequenceList: MutableList<SentenceSequence> = mutableListOf()
+      var currentIndex: Sentence? = null
+      var currentInSequence: MutableMap<Int, MutableList<Sentence>>? = null
+      sentencesForBooking.forEach {
+        if (it.consecutiveToSequence == null) {
```

Comment:
Just noting down here that, as discussed, this algorithm relies on the `sentenceSequence` values not interleaving between separate sequences within the same booking (e.g. you might have sequences [1,3,6] and [2,4,5]). We're waiting to hear back from the NOMIS team about this and, if that can happen, then we'll need to adjust for that.